pkg/controller/jobframework/tas_validation.go

```diff
@@ -65,7 +66,6 @@ func ValidateTASPodSetRequest(replicaPath *field.Path, replicaMetadata *metav1.O
 	}
 
 	// validate PodSetGroupName annotation
-	podSetGroupNameValue, podSetGroupNameFound := replicaMetadata.Annotations[kueuebeta.PodSetGroupName]
```

Comment:
why is this line moved? It is generally better to keep variable declaration close to usage

Reply:
Can we instead move the entire block 

```goalng
    podSetGroupNameValue, podSetGroupNameFound := replicaMetadata.Annotations[kueuebeta.PodSetGroupName]
	if podSetGroupNameFound {
		allErrs = append(allErrs, validatePodSetGroupNameAnnotation(podSetGroupNameValue, annotationsPath.Key(kueuebeta.PodSetGroupName))...)
	}
```
down, and merge with the new checks

Reply:
I moved it up because now it's used in two places (below the removed line and in the new code) so it made sense to me to move it closer to the rest of the annotations.

For example, `sliceSizeFound` is first used in line 81, but it's defined next to the rest of the annotation-related variables in line 36. It made sense to me to keep the `annotationValue, annotationFound` lines close together. WDYT?

Reply:
#7070